components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/dao/OAuthConsumerDAO.java

```diff
@@ -91,6 +95,49 @@ public String getOAuthConsumerSecret(String consumerKey) throws IdentityOAuthAdm
 
     }
 
+    /**
+     * Check whether the provided consumerKey, consumerSecret combination is exist or not in the database.
+     *
+     * @param consumerKey Consumer key.
+     * @param consumerSecret Consumer secret.
+     * @return Check the provided consumerKey, consumerSecret combination is exist or not.
+     * @throws IdentityOAuthAdminException Error when reading consumer key, consumer secret from the database.
+     */
+    public boolean isConsumerSecretExist(String consumerKey, String consumerSecret)
```

Comment:
Need to check the possibility of reusing getOAuthConsumerSecret() rather writing a new DAO..